apps/frontend/src/libs/components/balance-wheel/balance-wheel.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+	ArcElement,
+	Chart as ChartJS,
+	PolarAreaController,
+	RadialLinearScale,
+	Tooltip,
+} from "chart.js";
+
+import { useCallback, useEffect, useRef } from "~/libs/hooks/hooks.js";
+
+import { chartConfig } from "./libs/configs/configs.js";
+import { ANIMATION_INTERVAL } from "./libs/constants/constants.js";
+import { generateRandomData } from "./libs/helpers/generate-random-data.helper.js";
+import { type ChartDataType } from "./libs/types/types.js";
+import styles from "./styles.module.css";
+
+ChartJS.register(PolarAreaController, ArcElement, Tooltip, RadialLinearScale);
+
+type Properties = {
+	data?: ChartDataType[];
```

Comment:
`data` should be mandatory

Reply:
May I know why?I made data optional because the server might still be fetching the data and animate the wheel while fetching.

Reply:
For animation, the data will be initial data, which is generated by your `getRandomData` helper